src/hooks/usePublicRoomDirectory.ts

```diff
@@ -112,12 +114,14 @@ export const usePublicRoomDirectory = (): {
             }
 
             updateQuery(opts);
+            setLoading(true);
+            setError(undefined);
             try {
-                setLoading(true);
                 const { chunk } = await MatrixClientPeg.safeGet().publicRooms(opts);
                 updateResult(opts, showNsfwPublicRooms ? chunk : chunk.filter(cheapNsfwFilter));
                 return true;
             } catch (e) {
+                setError(e instanceof Error ? e : true);
```

Comment:
If the above code throws anything, shouldn't this set the error to true? Or can you add a doc comment explaining it?

Reply:
We store the error if it is an Error (catch type is `unknown`) otherwise `true` to signal there was an unknown error, this is documented above though quite tersely

Reply:
I misread the line. Sorry :see_no_evil: